### ISSUE TITLE
[NextJS] Editor main css being removed on Head change

### ISSUE THREAD

**Issue by relferreira** [suren-atoyan/monaco-react#272]
**Describe the bug**
It seems that when we change the favicon of a NextJS app, the `editor.main.css` link is removed from the `<head>`.

**To Reproduce**

1. Go to https://codesandbox.io/s/monaco-nextjs-gcd5d?file=/pages/index.js
2. Click twice on the button
3. The second time the editor is rendered without styles
4. If you comment the `<Head>` portion of the code, everything works fine

**Screenshots**

![image](https://user-images.githubusercontent.com/4078815/128177239-ee86ee39-429b-44a0-b3fa-c580ecdc6d68.png)

![image](https://user-images.githubusercontent.com/4078815/128177278-f72b89f5-bc8e-4803-8951-1f79aabc95a7.png)

**Desktop (please complete the following information):**
 - Browser Chrome
 - Version v4.2.1

**Comment by simerlec** [comment]
Can confirm that we're seeing the same issue - was about to open up a ticket myself when I saw this ❤️

**Comment by SpaceComet** [comment]
@relferreira  can you confirm something for me? I am having similar issue, not the same but I think it is cause by the same problem.

can you add this to your code:
```js
function handleEditorWillMount(monaco) {
        console.log("MOUTING");
    }

    function handleEditorDidMount(editor, monaco) {
        console.log("MOUNTED");
    }
```

and add this to your `Editor`
```js
beforeMount={handleEditorWillMount}
onMount={handleEditorDidMount}
```

In your case, every time you press the button the model is created again and again. 🤔 that definitely is creating some problems.
with the code I gave you, you can confirm that.

You can even have more problems if you work with multi-model like I am doing.

**Comment by relferreira** [comment]
Hi @SpaceComet 

I've updated the Codesandbox. Every time I press the button, both handlers are called.

**Comment by SpaceComet** [comment]
Yeah. it is making a new model. I just forked the code. I'm looking into it

**Comment by suren-atoyan** [comment]
> I've updated the Codesandbox. Every time I press the button, both handlers are called.

It is supposed to be like that; every time you press you unmount/mount the entire editor.

> In your case, every time you press the button the model is created again and again. 🤔 that definitely is creating some problems.

When you press the button, the editor is being mounted and it creates a new model, when you press one more time it's being unmounted and it removes the current model. Nothing special here.

@SpaceComet I don't know how creating a model can be related to this issue, but if you don't want to create-and-delete model during mounting/unmounting you can use `keepCurrentModel` property to use the same model across "different" editors.

Find `keepCurrentModel` [here](https://github.com/suren-atoyan/monaco-react#editor)

**Comment by suren-atoyan** [comment]
> If you comment the <Head> portion of the code, everything works fine

I guess this `<Head />` component does some "unfair" changes in the html-head that leads to this CSS inconsistency

**Comment by suren-atoyan** [comment]
[that's it](https://github.com/vercel/next.js/issues/11012)

**Comment by relferreira** [comment]
Thank you @suren-atoyan 
I "fixed" this issue adding the style manually
```
<link
rel="stylesheet"
type="text/css"
data-name="vs/editor/editor.main"
href="https://cdn.jsdelivr.net/npm/monaco-editor@0.25.2/min/vs/editor/editor.main.css"></link>
```

**Comment by SpaceComet** [comment]
@suren-atoyan  Thanks for taking the time. I am actually not sure I just thought that this problem is related to something I am experiencing. I'm making an example so I can show you. Once I have it Ill open a new issue to avoid confusion

**Comment by dimaslz** [comment]
Thank you @Glavin001 , @relferreira, and @suren-atoyan, this thread helped me a lot! :)
